core/utils/src/main/kotlin/edu/stanford/spezi/core/utils/extensions/SavedStateHandle.kt

```diff
@@ -0,0 +1,10 @@
+package edu.stanford.spezi.core.utils.extensions
+
+import androidx.lifecycle.SavedStateHandle
+import kotlinx.serialization.json.Json
+
+inline fun <reified T> SavedStateHandle.decode(key: String): T {
```

Comment:
Great.

 After looking at the usage of the tests where we need to stub the json, I think extension function would cause some repetitive code. What do you think of wrapping this in a custom component:
 
 ```kotlin
 @ViewModelScoped
class RouteParamProvider @Inject constructor(
    val savedStateHandle: SavedStateHandle,
) {
    inline fun <reified T> decode(key: String): T {
        val jsonString = savedStateHandle.get<String>(key) ?: throw IllegalArgumentException("Argument not found")
        return Json.decodeFromString(jsonString)
    }
}
 ```
 
 This way we will just need to stub the return value of the decode method and no need to setup a json in our tests